pkg/querier/queryrange/querysharding_test.go

```diff
@@ -234,6 +235,67 @@ func mockHandler(resp queryrange.Response, err error) queryrange.Handler {
 	})
 }
 
+func Test_InstantSharding(t *testing.T) {
+	ctx := user.InjectOrgID(context.Background(), "1")
+
+	var lock sync.Mutex
+	called := 0
+
+	sharding := NewQueryShardMiddleware(log.NewNopLogger(), queryrange.ShardingConfigs{
+		chunk.PeriodConfig{
+			RowShards: 3,
+		},
+	}, queryrange.NewInstrumentMiddlewareMetrics(nil),
+		nilShardingMetrics,
+		fakeLimits{
+			maxSeries: math.MaxInt32,
+			maxQueryParallelism: 10,
+		})
+	response, err := sharding.Wrap(queryrange.HandlerFunc(func(c context.Context, r queryrange.Request) (queryrange.Response, error) {
+		lock.Lock()
+		defer lock.Unlock()
+		called++
+		return &LokiPromResponse{Response: &queryrange.PrometheusResponse{
+			Data: queryrange.PrometheusData{
+				ResultType: loghttp.ResultTypeVector,
+				Result: []queryrange.SampleStream{
+					{
+						Labels:  []cortexpb.LabelAdapter{{Name: "foo", Value: "bar"}},
+						Samples: []cortexpb.Sample{{Value: 10, TimestampMs: 10}},
+					},
+				},
+			},
+		}}, nil
+	})).Do(ctx, &LokiInstantRequest{
+		Query:  `rate({app="foo"}[1m])`,
+		TimeTs: util.TimeFromMillis(10),
+		Path:   "/v1/query",
+	})
+	require.NoError(t, err)
+	require.Equal(t, 3, called, "expected 3 calls but got {}", called)
+	require.Len(t, response.(*LokiPromResponse).Response.Data.Result[0].Labels, 3)
+	require.Equal(t, &LokiPromResponse{Response: &queryrange.PrometheusResponse{
+		Status: "success",
+		Data: queryrange.PrometheusData{
+			ResultType: loghttp.ResultTypeVector,
+			Result: []queryrange.SampleStream{
+				{
+					Labels:  []cortexpb.LabelAdapter{
+						{Name: "foo", Value: "bar"},
+						{Name: "foo", Value: "bar"},
+						{Name: "foo", Value: "bar"},
+					},
+					Samples: []cortexpb.Sample{
+						{Value: 10, TimestampMs: 10},
+						{Value: 10, TimestampMs: 10},
+						{Value: 10, TimestampMs: 10},
+					},
+				},
+			},
+		},
```

Comment:
I assume we want three items in `SampleStream` instead. I couldn't trace down the conversion yet.